docs/notebooks/proton_radiography_particle_tracing_custom_source.ipynb

```diff
@@ -0,0 +1,524 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Proton Radiography: Synthetic Radiographs with Custom Source Profiles"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "An experimental reality of charged particle radiography experiments is that the source of the particles has finite size and often generates a non-uniform angular velocity distribution. For this reason, the`plasmapy.diagnostics.proton_radiography.SyntheticProtonRadiograph` particle tracing algorithm allows users to specify their own initial particle positions and velocities. This example will demonstrate how to use this functionality to create a more realistic synthetic radiograph that includes these source profile effects."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import astropy.constants as const\n",
+    "import astropy.units as u\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "import warnings\n",
+    "\n",
+    "from mpl_toolkits.mplot3d import Axes3D\n",
+    "\n",
+    "from plasmapy.diagnostics import proton_radiography as prad\n",
+    "from plasmapy.particles import Particle\n",
+    "from plasmapy.plasma.grids import CartesianGrid"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating Particles"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example we will create a distribution of 1e5 protons with a 5% variance in energy, a non-uniform angular velocity distribution, and a finite source extent."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nparticles = 1e5\n",
+    "particle = Particle(\"p+\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Creating the Initial Particle Velocities"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First we will create the velocity distribution by generating $\\theta$ and $\\phi$ angles for each particle. Generating $\\phi$ is simple: we will choose the azimuthal angles to just be uniformly distributed"
```

Comment:
This notebook does not have any mention on how the xyz coords are defined, so we don't have reference for `phi` and `theta`.  I'd also specifically state what `phi` and `theta` correspond to since they are often flipped between different research fields.

Reply:
I'll specify earlier that the source-detector axis is parallel to Z, and I'll explicitly define theta and phi as the polar and azimuthal angles respectively. I could consider a diagram, but that might be overkill?

Reply:
I'll leave that decision to you, but when it comes to defining coordinates it's always easier to understand the coord system via a diagram opposed to words.